packages/catalog-model/src/entity/ref.ts

```diff
@@ -160,12 +160,29 @@ export function parseEntityRef(
  * @param ref The reference to serialize
  * @returns The same reference on either string or compound form
  */
-export function serializeEntityRef(ref: {
-  kind?: string;
-  namespace?: string;
-  name: string;
-}): EntityRef {
-  const { kind, namespace, name } = ref;
+export function serializeEntityRef(
+  ref:
+    | Entity
+    | {
+        kind?: string;
+        namespace?: string;
+        name: string;
+      },
+): EntityRef {
+  let kind;
+  let namespace;
+  let name;
+
+  if ('apiVersion' in ref) {
```

Comment:
both are equally required so for the type check i just picked the top one :) but sure, from a readability standpoint it may seem more ontopic, can change